File: src/main/java/com/kairlec/contrller/RequestLogController.java
package com.kairlec.contrller;

import com.alibaba.fastjson.JSON;
import com.kairlec.exception.ServiceError;
import com.kairlec.local.utils.DownloadUtils;
import com.kairlec.local.utils.ResponseDataUtils;
import com.kairlec.utils.file.GetFileContent;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.*;
import java.util.ArrayList;
import java.util.List;

@RequestMapping("/request")
@RestController
public class RequestLogController {
    private static Logger logger = LogManager.getLogger(RequestLogController.class);

    @RequestMapping(value = "/content", produces = "text/plain; charset=utf-8")
    String get() {
        return ResponseDataUtils.successData("[" + GetFileContent.byPath("Log/request.log") + "]");
    }

    @RequestMapping(value = "/list", produces = "application/json; charset=utf-8")
    String list() {
        List<String> fileList = new ArrayList<>();
        File file = new File("Log/Request");
        if (file.exists()) {
            File[] files = file.listFiles();
            if (files != null) {
                for (File subFile : files) {
                    fileList.add(subFile.getName());
                }
            }
        }
        return ResponseDataUtils.successData(fileList);
    }

    @RequestMapping(value = "/download")
    String file(HttpServletRequest request, HttpServletResponse response) {
        try {
            return ResponseDataUtils.Error(DownloadUtils.log(request, response, "Log/Request"));
        } catch (IOException e) {
            e.printStackTrace();
            return ResponseDataUtils.Error(e);
        }
    }

}

File: src/main/java/com/kairlec/contrller/RequestLogController.kt
package com.kairlec.contrller

import com.kairlec.annotation.JsonRequestMapping
import com.kairlec.`interface`.ResponseDataInterface
import com.kairlec.local.utils.MultipartFileSender
import com.kairlec.local.utils.ResponseDataUtils.responseOK
import com.kairlec.local.utils.SKFileUtils
import com.kairlec.utils.content
import com.kairlec.utils.getSourcePath
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.io.File
import java.util.*
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

/**
 *@program: SKExplorer
 *@description: 请求日志记录器API
 *@author: Kairlec
 *@create: 2020-03-08 18:13
 */
@JsonRequestMapping(value = ["/request"])
@RestController
class RequestLogController {
    @RequestMapping(value = ["/content"])
    fun get() = "[" + File("Log/request.log").content() + "]".responseOK

    @RequestMapping(value = ["/list"])
    fun list(): ResponseDataInterface {
        val fileList: MutableList<String> = ArrayList()
        val file = File("Log/Request")
        if (file.exists()) {
            file.listFiles()?.let { files ->
                files.forEach { fileList.add(it.name) }
            }
        }
        return fileList.responseOK
    }

    @RequestMapping(value = ["/download"])
    fun file(request: HttpServletRequest, response: HttpServletResponse) = MultipartFileSender.fromPath(SKFileUtils.getLogPath("Log/Request", request.getSourcePath()), request, response).serveResource()
}